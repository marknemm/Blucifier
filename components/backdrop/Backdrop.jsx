import PropTypes from 'prop-types';
import Animated, { FadeIn, FadeOut } from 'react-native-reanimated';
import { useStyles } from './styles';

/**
 * The {@link Backdrop} component.
 *
 * @param {Types.Modal.BackdropProps} props The component {@link Types.Modal.BackdropProps properties}.
 * @returns {React.JSX.Element} The {@link Backdrop} component.
 */
export default function Backdrop({ isVisible = true, onPress, style }) {
  const styles = useStyles({ style });

  return isVisible && (
    <Animated.View
      entering={FadeIn}
      exiting={FadeOut}
      onTouchEnd={onPress}
      style={styles.backdrop}
    />
  );
}

Backdrop.propTypes = {
  isVisible: PropTypes.bool,
  onPress: PropTypes.func,
};
