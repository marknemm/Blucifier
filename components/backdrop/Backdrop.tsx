import Animated, { FadeIn, FadeOut } from 'react-native-reanimated';
import { Props } from './props';
import { useStyles } from './styles';

/**
 * Component for a modal backdrop.
 *
 * @param props The component {@link Props}.
 * @returns The {@link Backdrop} component.
 */
const Backdrop: React.FC<Props> = (props) => {
  const { isVisible = true, onPress } = props;
  const styles = useStyles(props);

  return isVisible && (
    <Animated.View
      entering={FadeIn}
      exiting={FadeOut}
      onTouchEnd={onPress}
      style={styles.backdrop}
    />
  );
};

export default Backdrop;
