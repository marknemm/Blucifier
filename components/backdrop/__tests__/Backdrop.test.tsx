import Backdrop from '@components/backdrop/Backdrop';
import AppProvider from '@test/contexts/app/AppProvider';
import { render } from '@testing-library/react-native';

describe('<Backdrop />', () => {
  describe('snapshots', () => {
    it('renders correctly', () => {
      const tree = render(
        <Backdrop />,
        { wrapper: AppProvider }
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders correctly when not visible', () => {
      const tree = render(
        <Backdrop isVisible={false} />,
        { wrapper: AppProvider }
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
