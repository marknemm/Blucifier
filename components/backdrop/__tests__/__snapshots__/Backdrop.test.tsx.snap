// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> snapshots renders correctly 1`] = `
<View
  collapsable={false}
  entering={[Function]}
  exiting={[Function]}
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
  style={
    {
      "backgroundColor": "#393e4233",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;

exports[`<Backdrop /> snapshots renders correctly when not visible 1`] = `null`;
